apps/src/templates/MultipleAssignButton.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import Button from './Button';
+import i18n from '@cdo/locale';
+import {
+  assignToSection,
+  unassignSection,
+  sectionsForDropdown
+} from '@cdo/apps/templates/teacherDashboard/teacherSectionsRedux';
+import MultipleSectionsAssigner from '@cdo/apps/templates/MultipleSectionsAssigner';
+import {sectionForDropdownShape} from '@cdo/apps/templates/teacherDashboard/shapes';
+
+class MultipleAssignButton extends React.Component {
+  static propTypes = {
+    sectionId: PropTypes.number.isRequired,
+    sectionName: PropTypes.string,
+    courseId: PropTypes.number,
+    courseOfferingId: PropTypes.number,
+    courseVersionId: PropTypes.number,
+    scriptId: PropTypes.number,
+    assignmentName: PropTypes.string,
+    reassignConfirm: PropTypes.func,
+    isOnCoursePage: PropTypes.bool,
+    isStandAloneUnit: PropTypes.bool,
+    participantAudience: PropTypes.string,
+    // Redux
+    assignToSection: PropTypes.func.isRequired,
+    isRtl: PropTypes.bool,
+    sectionsForDropdown: PropTypes.arrayOf(sectionForDropdownShape).isRequired
+  };
+
+  state = {
+    assignmentChoiceDialogOpen: false
+  };
+
+  onCloseDialog = () => {
+    this.setState({
+      assignmentChoiceDialogOpen: false
+    });
+  };
+
+  handleClick = () => {
+    this.setState({
+      assignmentChoiceDialogOpen: true
+    });
+  };
+
+  render() {
+    const {assignmentChoiceDialogOpen} = this.state;
+    const {
+      courseId,
+      courseOfferingId,
+      courseVersionId,
+      scriptId,
+      assignmentName,
+      isStandAloneUnit,
+      isRtl,
+      sectionsForDropdown,
+      participantAudience,
+      isOnCoursePage
+    } = this.props;
+
+    // Adjust styles if locale is RTL
+    const buttonMarginStyle = isRtl
+      ? styles.buttonMarginRTL
+      : styles.buttonMargin;
+
+    return (
+      <div>
+        <div style={buttonMarginStyle}>
+          <Button
+            __useDeprecatedTag
+            color={Button.ButtonColor.orange}
+            text={i18n.assignToMultipleSections()}
+            icon="plus"
+            onClick={this.handleClick}
+            className={'uitest-assign-button'}
+          />
+        </div>
+        {assignmentChoiceDialogOpen && (
+          <MultipleSectionsAssigner
+            assignmentName={assignmentName}
+            onClose={this.onCloseDialog}
+            sections={sectionsForDropdown}
+            courseId={courseId}
+            courseOfferingId={courseOfferingId}
+            scriptId={scriptId}
+            courseVersionId={courseVersionId}
+            reassignConfirm={this.props.reassignConfirm}
```

Comment:
nit: can you extract this with the rest of the props above?

Reply:
Done!